classes/PaymentModule.php

```diff
@@ -970,10 +1001,19 @@ protected function createOrderFromCart(
         $debug,
         $order_status,
         $id_order_state,
-        $carrierId = null
+        $carrierId = null,
```

Comment:
With the suggestions above, all this should be cleanable and the method remains untouched